src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -319,7 +319,8 @@ def create_archive_argv(
         )
 
         files_from = ("--files-from", input_file_list_filename) if input_file_list_filename else ()
-        return (self.path, f"c{compression}f", output_filename, *input_files) + files_from
+        # NB: -h follows symlinks, which ensures we don't capture a symlink to an immutable input, for instance
+        return (self.path, f"c{compression}fh", output_filename, *input_files) + files_from
```

Comment:
This is a tricky one. I would feel a little bit better about disabling the optimization in this case, because preserving symlinks (within `node_modules`, for example) is a huge benefit of your recent work to add symlink support. And tar'ing up a `node_modules` directory is likely to be common.

So... IMO, probably expose a "mutable paths" or "not symlinked" flag (as originally discussed on the ticket) on `Process` for this, and then the archive rules can enable/disable when they see `tar` (or a boolean field of `tar`)?

Reply:
Oh this might be painful. So for sure we should remove `-h` then, but getting this PR to pass without that might be challenging. I'll try it next week.

Reply:
Haha unfortunately I can't test this locally anymore because the cached value has the symlink. CI will hold true though.